src/eradiate_plugins/tests/bsdfs/test_hapke.py

```diff
@@ -0,0 +1,333 @@
+regression_test_geometries = [
+    (0.18430089, 1.46592582, 3.92553451, 0.39280281),
+]
+
+
+
+def angles_to_directions(theta, phi):
+    return mi.Vector3f(
+        np.sin(theta) * np.cos(phi), np.sin(theta) * np.sin(phi), np.cos(theta)
+    )
+
+def test_hapke_hotspot(variant_scalar_mono):
+    c = 0.273
+    c = (1 + c) / 2
+
+    hapke = mi.load_dict({ # Pommerol et al. (2013)
+        "type": "hapke",
+        "w": 0.526,
+        "theta": 13.3,
+        "b": 0.187,
+        "c": c,
+        "h": 0.083,
+        "B_0": 1.0,
+    })
+
+    theta_i = np.deg2rad(30.0)
+
+    theta_o = np.deg2rad(30.0)
+
+    phi_i = np.deg2rad(0.0)
+
+    phi_o = np.deg2rad(0.0)
+
+    wi = angles_to_directions(theta_i, phi_i)
+    wo = angles_to_directions(theta_o, phi_o)
+
+    values = np.asarray(eval_bsdf(hapke, wi, wo)) / np.abs(np.cos(theta_o))
+    assert np.allclose(values, 0.24746648)
+
+
+def test_hapke_grazing_viewing_angle(variant_scalar_mono):
+    c = 0.273
+    c = (1 + c) / 2
+
+    hapke = mi.load_dict({ # Pommerol et al. (2013)
+        "type": "hapke",
+        "w": 0.526,
+        "theta": 13.3,
+        "b": 0.187,
+        "c": c,
+        "h": 0.083,
+        "B_0": 1.0,
+    })
+
+    theta_i = np.deg2rad(30.0)
+    theta_o = np.deg2rad(-89.0)
+    phi_i = np.deg2rad(0.0)
+    phi_o = np.deg2rad(0.0)
+
+    wi = angles_to_directions(theta_i, phi_i)
+    wo = angles_to_directions(theta_o, phi_o)
+
+    values = np.asarray(eval_bsdf(hapke, wi, wo)) / np.abs(np.cos(theta_o))
+    assert np.allclose(values, 0.15426355)
+
+
+def test_hapke_backward(variant_scalar_mono):
+    c = 0.273
+    c = (1 + c) / 2
+
+    hapke = mi.load_dict({ # Pommerol et al. (2013)
+        "type": "hapke",
+        "w": 0.526,
+        "theta": 13.3,
+        "b": 0.187,
+        "c": c,
+        "h": 0.083,
+        "B_0": 1.0,
+    })
+
+    theta_i = np.deg2rad(30.0)
+    theta_o = np.deg2rad(80.0)
+    phi_i = np.deg2rad(0.0)
+    phi_o = np.deg2rad(0.0)
+
+    wi = angles_to_directions(theta_i, phi_i)
+    wo = angles_to_directions(theta_o, phi_o)
+
+    values = np.asarray(eval_bsdf(hapke, wi, wo)) / np.abs(np.cos(theta_o))
+    assert np.allclose(values, 0.19555340)
+
+
+def test_hapke_principal_plane(variant_llvm_rgb):
+
+    c = 0.273
+    c = (1 + c) / 2
+
+    hapke = mi.load_dict({ # Pommerol et al. (2013)
+        "type": "hapke",
+        "w": 0.526,
+        "theta": 13.3,
+        "b": 0.187,
+        "c": c,
+        "h": 0.083,
+        "B_0": 1.0,
+    })
+
+    theta_i = np.deg2rad(30.0 * np.ones((180,)))
+    theta_o = np.deg2rad(np.arange(90, -90, -1))
+    phi_i = np.deg2rad(np.zeros((180,)))
+    phi_o = np.deg2rad(np.zeros((180,)))
+
+    wi = angles_to_directions(theta_i, phi_i)
+    wo = angles_to_directions(theta_o, phi_o)
+
+    values = np.asarray(eval_bsdf(hapke, wi, wo)) / np.abs(np.cos(theta_o))
+
+    fig = plt.figure()
+
+    plt.plot(np.rad2deg(theta_o), values)
+    plt.grid()
+
+    plt.savefig("test.png")
+    plt.close()
```

Comment:
Please don't have the test output a plot by default, this slows down execution a lot.